core/wasmJs/src/JSJodaExceptions.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019-2020 JetBrains s.r.o.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime
+
+
+private fun checkExceptionName(exception: JsAny, name: String): Boolean =
+    js("exception.name === name")
+
+private fun checkForName(exception: Throwable, name: String): Boolean {
+    val cause = (exception as? JsException)?.jsException ?: return false
+    return checkExceptionName(cause, name)
+}
+
+internal actual fun Throwable.isJodaArithmeticException(): Boolean = checkForName(this, "ArithmeticException")
+internal actual fun Throwable.isJodaDateTimeException(): Boolean = checkForName(this, "DateTimeException")
+internal actual fun Throwable.isJodaDateTimeParseException(): Boolean = checkForName(this, "DateTimeParseException")
+
+private fun withCaughtJsException(body: () -> Unit): JsAny? = js("""{
+    try {
+        body();
+        return null;
+    } catch(e) {
+        return e;        
+    }
+}""")
+
+internal class JsException(val jsException: JsAny): Throwable()
```

Comment:
This exception needs more care.
* It will be the `cause` of most of the exceptions we throw.
* If JS throws anything unusual, something we don't process, the user won't have any idea at all what happened.

For example, when I attempt to construct `LocalDateTime(2023, 2, 30)`, I get `IllegalArgumentException: JsException` on wasm.

At the very least, the exception message should be preserved. If it's easy to also preserve the stacktrace, it would be nice, but it's not critical and I expect it not to be simple.

Reply:
Now, in WasmJs, we got `JsException` which is caused from joda and not processed by DateTime. The message now is preserved from joda exception. The callstack a bit mess and there is no good way for now to put original call stack from joda side into `JsException` (we can make an extra method for `JsException` that receive the original stack trace from `jsException` field)